apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -109,4 +152,77 @@ defmodule GameBackend.CurseOfMirra.Quests do
       Repo.transaction(multi)
     end
   end
+
+  def reroll_quest(daily_quest_id) do
+    reroll_configurations = Application.get_env(:game_backend, :quest_reroll_config)
+
+    daily_quest =
+      Quests.get_daily_quest(daily_quest_id)
+
+    amount_of_rerolled_daily_quests =
+      Quests.get_user_today_rerolled_daily_quests_by_type(daily_quest.user_id, daily_quest.quest.type)
+      |> Enum.count()
+
+    reroll_costs =
+      reroll_configurations.costs
+      |> Enum.map(fn currency_cost_params ->
+        currency =
+          Currencies.get_currency_by_name_and_game!(
+            currency_cost_params.quest_currency_cost_name,
+            Utils.get_game_id(:curse_of_mirra)
+          )
+
+        amount =
+          (currency_cost_params.quest_base_cost +
+             currency_cost_params.quest_base_cost * amount_of_rerolled_daily_quests *
+               reroll_configurations.reroll_multiplier)
+          |> round()
+
+        %CurrencyCost{currency_id: currency.id, amount: amount}
+      end)
+
+    get_user_missing_quests_by_type(daily_quest.user_id, daily_quest.quest.type)
+    |> case do
+      [] ->
+        {:error, :not_enough_available_quests}
+
+      quests ->
+        new_quest =
+          quests
+          |> Enum.random()
+
+        attrs = %{
+          user_id: daily_quest.user_id,
+          quest_id: new_quest.id,
+          status: "available"
+        }
+
+        new_quest_changeset = DailyQuest.changeset(%DailyQuest{}, attrs)
+
+        finish_previous_quest_changeset =
+          DailyQuest.changeset(daily_quest, %{
+            status: "rerolled"
+          })
+
+        cond do
+          daily_quest.status == "rerolled" ->
+            {:error, :quest_rerolled}
+
+          not Currencies.can_afford(daily_quest.user_id, reroll_costs) ->
+            {:error, :cant_afford}
+
+          true ->
+            Multi.new()
+            # Deduct currency
+            |> Multi.run(:deduct_currencies, fn _, _ ->
+              Currencies.substract_currencies(daily_quest.user_id, reroll_costs)
+            end)
+            # Update old daily quest
+            |> Multi.update(:change_previous_quest, finish_previous_quest_changeset)
+            # Add new daily quest
+            |> Multi.insert(:insert_quest, new_quest_changeset)
+            |> Repo.transaction()
```

Comment:
The comments are redundant to the multi operation description, maybe you could replace the description with the comment (then delete the comment) to be more explicit and accurate.

Reply:
I just deleted them, the multi key should be enough to know what's happening imo